rust/index/src/fulltext/types.rs

```diff
@@ -101,13 +98,8 @@ impl<'me> FullTextIndexWriter<'me> {
                     // Readers are uninitialized until the first compaction finishes
                     // so there is a case when this is none hence not an error.
                     None => 0,
-                    Some(reader) => {
-                        match reader.get_frequencies_for_token(token.text.as_str()).await {
-                            Ok(frequency) => frequency,
-                            // New token so start with frequency of 0.
-                            Err(_) => 0,
-                        }
-                    }
+                    Some(reader) => (reader.get_frequencies_for_token(token.text.as_str()).await)
```

Comment:
huge style nit: I prefer unwrap_or(0) since it may not be working knowledge of the reader what the default is